AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -753,7 +753,9 @@ - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSe
 
 - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout sizeForItemAtIndexPath:(NSIndexPath *)indexPath
 {
-  return [[self nodeForItemAtIndexPath:indexPath] calculatedSize];
+  ASDisplayNode *node = [self nodeForItemAtIndexPath:indexPath];
+  [node layoutIfNeeded];
```

Comment:
layoutIfNeeded doesn't seem ideal to do here as it will visit many subnodes to set frames - grabbing locks, etc.  For tableView for example that occurs for all the rows in the datasource...all at once, a huge number.

Since this is internal code -- I think we need to get the pendingLayout's calculatedSize.  We could add a +FrameworkPrivate method for "latestCalculatedSize" or just update the implementation of -calculatedSize to return that.

I actually think it makes sense to have node.calculatedLayout return the pendingLayout if it exists, so that would solve this problem.  Maybe we should offer an external property .currentLayout or .appliedLayout. E.g.:

.calculatedLayout = _pendingLayout ? : _currentLayout;
.appliedLayout = _currentLayout; // even if nil

I bet there is not a use case for .appliedLayout currently.  So I think it is safe to change .calculatedLayout to return _pendingLayout if it exists, and then this code can just call -calculatedSize as it did before.  We can add .appliedLayout later if a use case comes up.